packages/blocks/src/api/utils.js

```diff
@@ -275,3 +275,24 @@ export function __experimentalSanitizeBlockAttributes( name, attributes ) {
 		{}
 	);
 }
+
+/**
+ * I created this wrapper to hide the complexity for the consumer..
+ *
+ * @param {*} name
+ * @param {*} role
+ */
+// TODO jsdoc
+// TODO tests
+export function __experimentalGetBlockAttributesNamesByRole( name, role ) {
+	const attributes = getBlockType( name )?.attributes;
+	if ( ! attributes ) return;
+	if ( ! role ) return Object.keys( attributes );
+	return Object.entries( attributes ).reduce(
+		( accumulator, [ attributeName, schema ] ) => {
+			if ( schema?.role === role ) accumulator.push( attributeName );
+			return accumulator;
+		},
+		[]
+	);
```

Comment:
Totally minor point, but this may be clearer if based on `keys / filter`:

```suggestion
	return Object.keys( attributes ).filter(
		( attributeName ) => attributes[ attributeName ]?.role === role
	);
```

Reply:
It seems I need to reduce `reduce` usage :) - it's clearer the way you suggested.